models/users.js

```diff
@@ -572,6 +572,59 @@ const fetchAllUsers = async () => {
   return users;
 };
 
+const archiveUserIfNotInDiscord = async () => {
```

Comment:
I don't think this should be done in models, here in a single function we are getting and updating, please make separate concerns.